features/call/src/test/kotlin/io/element/android/features/call/ui/CallScreenPresenterTest.kt

```diff
@@ -0,0 +1,238 @@
+ */
+
+package io.element.android.features.call.ui
+
+import app.cash.molecule.RecompositionMode
+import app.cash.molecule.moleculeFlow
+import app.cash.turbine.test
+import com.google.common.truth.Truth.assertThat
+import io.element.android.features.call.CallType
+import io.element.android.features.call.utils.CallWidgetProvider
+import io.element.android.features.call.utils.WidgetMessageInterceptor
+import io.element.android.libraries.architecture.Async
+import io.element.android.libraries.core.coroutine.CoroutineDispatchers
+import io.element.android.libraries.matrix.api.core.RoomId
+import io.element.android.libraries.matrix.api.core.SessionId
+import io.element.android.libraries.matrix.api.widget.MatrixWidgetDriver
+import io.element.android.libraries.matrix.test.A_ROOM_ID
+import io.element.android.libraries.matrix.test.A_SESSION_ID
+import io.element.android.libraries.matrix.test.widget.FakeWidgetDriver
+import io.element.android.libraries.network.useragent.UserAgentProvider
+import io.element.android.services.toolbox.api.systemclock.SystemClock
+import io.element.android.tests.testutils.WarmUpRule
+import io.element.android.tests.testutils.testCoroutineDispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+
+class CallScreenPresenterTest {
+
+    @get:Rule
+    val warmUpRule = WarmUpRule()
+
+    private var getWidgetCalled = false
+
+    @Test
+    fun `present - with CallType ExternalUrl just loads the URL`() = runTest {
+        val presenter = createCallScreenPresenter(CallType.ExternalUrl("https://call.element.io"))
+        moleculeFlow(RecompositionMode.Immediate) {
+            presenter.present()
+        }.test {
+            // Wait until the URL is loaded
+            skipItems(1)
+
+            val initialState = awaitItem()
+            assertThat(initialState.urlState).isEqualTo(Async.Success("https://call.element.io"))
+            assertThat(initialState.isInWidgetMode).isFalse()
+        }
+    }
+
+    @Test
+    fun `present - with CallType RoomCall loads URL and runs WidgetDriver`() = runTest {
+        val widgetDriver = FakeWidgetDriver()
+        val presenter = createCallScreenPresenter(
+            callType = CallType.RoomCall(A_SESSION_ID, A_ROOM_ID),
+            widgetDriver = widgetDriver,
+        )
+        moleculeFlow(RecompositionMode.Immediate) {
+            presenter.present()
+        }.test {
+            // Wait until the URL is loaded
+            skipItems(1)
+
+            val initialState = awaitItem()
+            assertThat(initialState.urlState).isInstanceOf(Async.Success::class.java)
+            assertThat(initialState.isInWidgetMode).isTrue()
+            assertThat(getWidgetCalled).isTrue()
```

Comment:
Maybe set `getWidgetCalled` to `false` at the beginning of the test. It's not used in other test, so it's fine, but it would be more correct.

Reply:
I think every test will be run using their own instance of the class, but I can create a `@Before` function for that anyway.